### ISSUE TITLE
feature: Expose all information related to a Proposed Change via GraphQL subscriptions

### ISSUE THREAD

**Issue by dgarros** [opsmill/infrahub#863]
### Component

API Server / GraphQL

### Describe the Feature Request

A Proposed Change centralise a lot of information about a branch like comment, checks, artifacts etc .. and some of these data can change very rapidly.

To allow the frontend to provide a more reactive interface and to report the latest changes as quickly as possible to the user it would be good to expose all data related to a Proposed Change via a GraphQL Subscriptions. 

### Describe the Use Case

Improve the user experience and display new update as far as possible in the frontend

**Comment by tobiasjohanssonsdnit** [comment]
can this be for external graphql/InfrahubSDK clients as well or just Infrahub frontend?

**Comment by ogenstad** [comment]
Yes, the frontend will not be using any custom API calls that isn't available to the public. The main frontend specific stuff in Infrahub will be things related to the users menus and other design elements that typically won't be relevant for external entities (the information will of course be accessible, just that it might not provide a lot of value for others).

Other than that anything that the frontend can do will be available using the same API.
